### PR TITLE
Support same padding for keras atrousconv1d,2d and conv3d

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Support same padding for keras atrousconv1d,2d and conv3d

## How was this patch tested?

Jenkins

### CODE REVIEW

File: pyspark/bigdl/keras/converter.py [hunk 1]
```diff
@@ -1030,21 +1025,30 @@ def create_atrousconvolution2d(self, klayer, kclayer):
...
+        w = input_shape[3]
+        kh = config["nb_row"]
+        kw = config["nb_col"]
+        dh = config["subsample"][0]
+        dw = config["subsample"][1]
+        dilation_h = config["atrous_rate"][0]
+        dilation_w = config["atrous_rate"][1]
+        pad_h = 0
+        pad_w = 0
+        if config["border_mode"] == "same":
```
Comment (zhichao-li) [inline pyspark/bigdl/keras/converter.py]: extract this as a function?

File: pyspark/bigdl/keras/converter.py [hunk 2]
```diff
@@ -886,16 +888,6 @@ def to_bigdl_2d_padding(self, border_mode):
         else:
             raise Exception("Unsupported border mode: %s" % border_mode)
 
-    def to_bigdl_1d_padding(self, border_mode, kernel_w):
-        if border_mode == "same":
```
Comment (zhichao-li) [inline pyspark/bigdl/keras/converter.py]: why delete this?

File: pyspark/bigdl/keras/converter.py (hunk 1 above)
Comment (hkvision) [reply pyspark/bigdl/keras/converter.py]: I consider extracting it as a function, but this function will need too many arguments but only does one calculation?

File: pyspark/bigdl/keras/converter.py (hunk 2 above)
Comment (hkvision) [reply pyspark/bigdl/keras/converter.py]: This is never used lol. For all 1D layers, we use 2D layers to achieve. Thus, for padding, the manner is the same as 2D.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V-P/job/BigDL-PR-V/229/

**Comment by hkvision** [comment]
Have a look on the calculation of padding? @yangw1234 Any comments or corner cases missing? Thanks

**Comment by yangw1234** [comment]
LGTM

**Comment by hkvision** [comment]
@yangw1234 thanks for the review!

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-V-Spark-1.6/473/

**Pull request merged**